### ISSUE TITLE
8. Develop functionality to download taxon lists (csv and pdf)

### ISSUE THREAD

**Issue by helendallas** [kartoza/django-bims#3951]
Currently it is possible to download a csv and xls file.

@schmikloi to provide details of the  download file needed by the experts, including

what's columns to include

**Comment by schmikloi** [comment]
csv and  xls are fine, but a pdf file will be needed as well; it could look similar to this one:
[Crustacea-Ostracoda FADA generated checklist.pdf](https://github.com/user-attachments/files/15592625/Crustacea-Ostracoda.FADA.generated.checklist.pdf)

I will provide an exact template at some stage, but this issue has NO priority for now

**Comment by dimasciput** [comment]
I have added the size for this ticket, this might take 1-2 days to implement.

**Comment by helendallas** [comment]
As per our meeting, priority is to improve csv download and develop pdf download functionality.

**Comment by dimasciput** [comment]
the pdf download has been added to staging

@helendallas @schmikloi

**Comment by helendallas** [comment]
@schmikloi Please can you check this, thanks.

**Comment by helendallas** [comment]
Also in my test upload of Plecoptera South Africa, the pdf included INDIGENOUS for all taxa, although this was not in my csv upload file. Where has this come from?

<img width="632" alt="Image" src="https://github.com/user-attachments/assets/b79e2fa1-88eb-4ddc-9ed9-a3a55412e6aa" />

**Comment by helendallas** [comment]
@schmikloi Dimas is working on FADA today, so please try and test and give feedback soonest

**Comment by schmikloi** [comment]
> [@schmikloi](https://github.com/schmikloi) Dimas is working on FADA today, so please try and test and give feedback soonest

I try what I can (despite being not well), but the focus now has to be on the correct upload, validation etc.; the experts want to start working; I will address all the download options as well as the API stuff afterwards

**Comment by helendallas** [comment]
@schmikloi No problem, sorry you are not well. Agreed, lets get the upload etc finished first.

**Comment by schmikloi** [comment]
@dimasciput 
I checked the current csv download; the following things are required:
- for the public user only validated species can be exported (as they cannot see the unvalidated species anyway, this should be ok)
- if there is a filter, only filtered taxa should be exported --> works already
- fields that should be exported (in this order): Kingdom, Phylum, Class, Order, Family, Subfamily, Tribe, Subtribe, Genus, Subgenus, Species group, Species, Subspecies, Taxon Name, Author(s), Status, Accepted Taxon, Rank, Taxonomic Comments, Taxonomic References, Biogeographic distribution, Biogeographic Comments, Biogeographic References, Environmental Information, Environmental Comments, Environmental References, Conservation Status, Conservation Comments, Conservation References, Common Name, GBIF link, FADA ID
- currently a lot more fields are exported and some others are missing
- currently the tags are not exported correctly and the headlines are partly missing
<img width="235" alt="Image" src="https://github.com/user-attachments/assets/1b33e211-d57a-41a3-a9c9-2b24fd5fbb77" />

**Comment by schmikloi** [comment]
@dimasciput @helendallas 
Just thinking about the pdf, I think the csv should also contain a version number, a download date and a citation

As to the latter, we can either add a field somewhere for each organism group where we type in the citation or we generate it automatically from team leader plus team members and the organism group...
How do you handle this for the SA species lists?

**Comment by schmikloi** [comment]
@dimasciput 
ok, the download works including the time stamp, that is fine

the following changes are necessary though:
- as stated above, we need a proper citation; this could for example look like:

<img width="1007" height="94" alt="Image" src="https://github.com/user-attachments/assets/e7ba16e5-8a6f-45f5-8b6d-a7ee5dd031cc" />

you could create a table in the back that we fill with the different parts of the citation and you then create it from there

- additionally, the exported columns need to be adapted to our imported columns (i.e. you can remove origin, endemism, invasion, etc. but need to add columns like our comments etc.)

**Comment by dimasciput** [comment]
Hi @schmikloi 

> you could create a table in the back that we fill with the different parts of the citation and you then create it from there

So, should I create a citation table that has a relationship with the taxon group or with the taxonomy? Also, could you tell me what the contents of the table should be? Perhaps you could create a blank template table in Excel.

**Comment by dimasciput** [comment]
@schmikloi Yes, I think a table will be easier for me to implement, but I'm still not sure who will populate it. Will the experts upload a CSV file (in which case I’ll need to create an uploader for them), or will the admin fill it manually from the admin page (which would only require me to create the table in the backend, and that should be quick)?

And just to confirm, the table will basically contain the author name, source name, and taxon group, right?

**Comment by schmikloi** [comment]
@dimasciput 
this table will be populated by the admin, yes
the table should contain the organism group and the authors as well as a text for the citation, e.g. "Ostracoda Checklist; Freshwater Animal Diversity Assessment (FADA)"; the latter should basically be the same for all organism groups but it might be that we still change it from what I suggest here to something different; having it in the table would allow us to change the citations any time without having to bother you
year and access date need to be added automatically

does this make sense to you?

**Comment by dimasciput** [comment]
@schmikloi Yes, I'm going to create the table now, then I'll ask you to check it. After that, I'll update the CSV with the data from this table.

**Comment by dimasciput** [comment]
@schmikloi here's the table https://fada.kartoza.com/admin/bims/taxongroupcitation/, you should be able to add some data.
And I also have updated the csv and pdf download to include the citations.

**Comment by dimasciput** [comment]
@schmikloi I found another issue on the csv with regard this data

| On GBIF | GBIF URL | Origin | Endemism | FADA ID | Taxon Rank | Kingdom | Phylum | Class | Order | Family | SubFamily | Tribe | SubTribe | Genus | SubGenus | SpeciesGroup | Species | SubSpecies | Author(s) | Taxonomic Status | Accepted Taxon | Taxonomic Reference (Valid) | Taxonomic Comments | ANT | AT | AU | NA | NT | OL | PA | PAC | Biogeographic References | Biogeographic Comments | aquatic | water-dependent | freshwater | limno-terrestrial | brakish | others | lentic | lotic | parasitic | hosts | Environmental References | Environmental Comments | Common Name | Conservation Status | Conservation Reference | Conservation Comments |
|---------|----------|--------|----------|---------|------------|---------|--------|-------|-------|--------|-----------|-------|----------|-------|----------|--------------|---------|------------|-----------|------------------|----------------|-----------------------------|--------------------|-----|----|----|----|----|----|----|-----|--------------------------|-------------------------|---------|-----------------|------------|-------------------|---------|--------|--------|------|-----------|-------|--------------------------|-------------------------|-------------|------------------|--------------------------|-----------------------|
| Yes | 9660818 |  |  |  | Species | Animalia | Nematomorpha | Gordioida | Gordiida | Chordodidae | Chordodinae |  |  | Gordius |  |  | De Filippii |  | Rosa, 1881 | Synonym | Chordodes defilippi |  |  |  |  |  |  |  |  |  | Y |  | Y |  |  |  |  |  |  |  |  |  |  |  |  |  |


The GBIF URL points to a genus, but this is actually a species. That’s why the system got confused and created some incorrect data for everything. I can fix this quickly from the backend, but it’s better to review and correct the data itself. I already have a lot of validation in the uploader, and if we keep adding more checks, it may break other things.

**Comment by dimasciput** [comment]
@schmikloi ok I fixed the data by set the On GBIF to No and remove the GBIF URL, please try download the data again.

**Comment by schmikloi** [comment]
@dimasciput this seems to work properly now; but I am not sure if I understand what was the issue; I need to tell the experts what thy should do better, so maybe you can explain this at some stage
